Job creator elevation (#9420)
fishtown-analytics/docs.getdbt.com: website/docs/docs/platform/manage-access/enterprise-permissions.md

Notable features:
- Job creator is a project-level set.

> - Can create, edit, and run jobs in environment types where the group has [environment write access](/docs/platform/manage-access/environment-permissions-setup). Without write access, job and run actions are read-only.

Reviewer: I don't think this part is how the Job Creator role operates. It's specifically for creating jobs and can do so without any environment write access assigned as long as it belongs to a project the group has access to
```suggestion
- Can create, edit, and run jobs within assigned projects and environments.
```
